dp/cloud/python/magma/configuration_controller/tests/unit/test_crl_validator.py

```diff
@@ -0,0 +1,149 @@
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import time
+from typing import Dict, List
+from unittest import TestCase, mock
+
+import requests_mock
+from freezegun import freeze_time
+from magma.configuration_controller.crl_validator.crl_validator import (
+    CRLValidator,
+)
+from magma.configuration_controller.request_router.request_router import (
+    RequestRouter,
+)
+from magma.configuration_controller.tests.unit.fixtures.crl.certs import (
+    CERTIFICATES_DATE,
+    NO_CRL_CERT,
+    REVOKED_CERT_CRLS_DATA,
+    REVOKED_CRL_CERT,
+    VALID_CERT_CRLS_DATA,
+    VALID_CRL_CERT,
+)
+from magma.mappings.request_mapping import request_mapping
+from mocket import mocket
+from parameterized import parameterized
+from requests.exceptions import SSLError
+
+
+@requests_mock.Mocker()
+@freeze_time(CERTIFICATES_DATE)
+class CRLValidatorTestCase(TestCase):
+    sas_url = 'https://fake.sas.url/'
+
+    @parameterized.expand([
+        (VALID_CRL_CERT, VALID_CERT_CRLS_DATA),
+        (NO_CRL_CERT, {}),
+    ])
+    @mocket.mocketize
+    def test_valid_certificates_passes(
+            self,
+            mocker: requests_mock.Mocker,
+            certificate: bytes,
+            crls: Dict[str, bytes],
+    ) -> None:
+        # Given
+        self._set_mocket_cert(certificate=certificate)
+        self._set_mocker_CRL(mocker=mocker, crls=crls)
+        validator = self._create_validator(urls=[self.sas_url])
+
+        # When
+        valid = validator.is_valid(url=self.sas_url)
+
+        # Then
+        self.assertIs(valid, True)
+
+    @mocket.mocketize
+    def test_exception_is_raised_on_revoked_certificate(self, mocker: requests_mock.Mocker) -> None:
+        # Given
+        self._set_mocket_cert(certificate=REVOKED_CRL_CERT)
+        self._set_mocker_CRL(mocker=mocker, crls=REVOKED_CERT_CRLS_DATA)
+        validator = self._create_validator(urls=[self.sas_url])
+
+        # When / Then
+        with self.assertRaises(SSLError):
+            validator.is_valid(url=self.sas_url)
+
+    @mocket.mocketize
+    def test_certificates_are_updated_in_the_background(self, mocker: requests_mock.Mocker) -> None:
+        # Given
+        self._set_mocket_cert(certificate=VALID_CRL_CERT)
+        self._set_mocker_CRL(mocker=mocker, crls=VALID_CERT_CRLS_DATA)
+
+        update_rate = 1  # seconds
+        validator = self._create_validator(urls=[self.sas_url], update_rate=update_rate)
+        valid = validator.is_valid(url=self.sas_url)
+        self.assertIs(valid, True)
+
+        # When
+        self._set_mocket_cert(certificate=REVOKED_CRL_CERT)
+        self._set_mocker_CRL(mocker=mocker, crls=REVOKED_CERT_CRLS_DATA)
```

Comment:
can we also check that after setting those certs nothing happened yet (cert is still valid)?